### ISSUE TITLE
Armorstand Projectile bug

### ISSUE THREAD

**Issue by Nightshade73** [WeaponMechanics/MechanicsMain#124]
### Has this issue already been reported? Did you ask for help before making this report?

- [X] I have searched the existing issues
- [X] I have asked for help

### Current Behavior

'Fake invisible armor stands' don't have the same packet output as 'Real invisible armor stands'.

The error was noticed while using     Fully_Automatic_Shots_Per_Second: <Integer> 
while also
`  Projectile:
    Projectile_Settings:
      Type: "ARMOR_STAND"`
      
     Armor stands block right click actions (shooting), so while using a fast firerate the projectile Armor Stands hinder firing causing the Full_Auto_Shots to act like a burst fire.
     
  

### Expected Behavior

The Fully_Automatic_Shots_Per_Second: <Integer>  module shouldn't be hindered by using Armor_Stand projectiles.

### Steps To Reproduce

```YML
AK47:
  Info:
    Weapon_Item:
      Type: "CARROT_ON_A_STICK"
      Name: "&eAK47"
      Lore:
        - "&7Very reliable automatic assault rifle chambered in 7.62mm Soviet."
        - "&7Large caliber ammunition requires skill to control recoil."
        - ""
        - "&7Weapon Statistics &6-&7 66.5"
        - "&6➣&7 68 &6⟸ &7Accuracy"
        - "&6➣&7 76 &6⟸ &7Damage"
        - "&6➣&7 67 &6⟸ &7Range"
        - "&6➣&7 61 &6⟸ &7Fire Rate"
        - "&6➣&7 59 &6⟸ &7Mobility"
        - "&6➣&7 68 &6⟸ &7Control"
      Unbreakable: true
      Hide_Flags: true
      Attributes:
        - "GENERIC_MOVEMENT_SPEED--0.0181"
      Deny_Use_In_Crafting: true
    Weapon_Info_Display:
      Action_Bar:
        Message: "&eAK47%firearm-state% &7«&6%ammo-left%&7»&6%reload%"
    Weapon_Get_Mechanics:
      Sounds:
        - "ITEM_ARMOR_EQUIP_CHAIN-1-0.75"
    Weapon_Equip_Delay: 30
    Show_Cooldown:
      Weapon_Equip_Delay: true
      Reload_Time: true
    Cancel:
      Drop_Item: true
      Arm_Swing_Animation: true
  Skin:
    Default:
      Type: CARROT_ON_A_STICK
      Custom_Model_Data: 5
    Reload:
      Type: CARROT_ON_A_STICK
      Durability: 5
    Scope:
      Type: CARROT_ON_A_STICK
      Custom_Model_Data: 6
  Projectile:
    Projectile_Settings:
      Type: "ARMOR_STAND"
      Projectile_Item_Or_Block:
        Type: "BLAZE_ROD"
        Custom_Model_Data: 1
      Minimum:
        Speed: 60
      Gravity: 0.009
  Shoot:
    Trigger:
      Main_Hand: "RIGHT_CLICK"
      Off_Hand: "RIGHT_CLICK"
      Deny_When:
        Swimming: true
    Projectile_Speed: 100
    Projectiles_Per_Shot: 1
    Fully_Automatic_Shots_Per_Second: 10
    Spread:
      Base_Spread: 0.01
      Modify_Spread_When:
        Zooming: 50%
        Sneaking: 50%
        In_Midair: 150%
    Recoil:
      Push_Time: 0
      Recover_Time: 0
      Horizontal:
        - 1.0
        - 0
      Vertical:
        - 3.19
      Modify_Recoil_When:
        Zooming: 50%
        Sneaking: 50%
        In_Midair: 150%
    Mechanics:
      Sounds:
        - "custom:shoot.rifle6-16-1-0-0.1"
        #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
  Reload:
    Trigger:
      Main_Hand: "DROP_ITEM"
      Off_Hand: "DROP_ITEM"
    Magazine_Size: 30
    Reload_Duration: 34
    Start_Mechanics:
      Sounds:
        - "custom:reload.start.normal-0.5-1-0-0.1"
        #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
    Finish_Mechanics:
      Sounds:
        - "custom:reload.end.normal-0.5-1-0-0.1"
        #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
  Damage:
    Base_Damage: 5
    Armor_Damage: 4
    Dropoff:
      - "50~1.00"
      - "100~2.00"
  Scope:
    Trigger:
      Main_Hand: "LEFT_CLICK"
      Off_Hand: "LEFT_CLICK"
    Zoom_Amount: 8
    Mechanics:
      Sounds:
        - "custom:scope.in-0.5-1-0-0.1"
        #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
    Zoom_Off:
      Mechanics:
        Sounds:
          - "custom:scope.out-0.5-1-0-0.1"
          #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
  Firearm_Action:
    Type: "SLIDE"
    Open:
      Time: 8
      Mechanics:
        Sounds:
          - "BLOCK_NOTE_BLOCK_HAT-0.5-1-0-0.1"
          - "BLOCK_PISTON_CONTRACT-0.5-1-2-0.1"
          #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
    Close:
      Time: 8
      Mechanics:
        Sounds:
          - "BLOCK_PISTON_EXTEND-0.5-1-0-0.1"
          #Sound-volume(chunks)-Pitch(0.5-2)-Delay-PitchNoise
```

### Frequency

100% reoccurrence 

### Version and Log

![2022-06-25_22 33 20](https://user-images.githubusercontent.com/23036913/175798297-e1fcbc12-bdd8-473e-a494-0070c6229077.png)
[latest.log](https://github.com/WeaponMechanics/MechanicsMain/files/8985931/latest.log)

**Comment by DeeCaaD** [comment]
`PlayerInteractEntityEvent` isn't called for FakeEntitites, this might need packet listener

**Comment by CJCrafter** [comment]
https://youtu.be/dE5miivFTSU

**Comment by DeeCaaD** [comment]
So the main issue where right-click can't be held properly will be fixed when #140 is merged, so I can just directly code it using ProtocolLib.  

Ill look at the other issue where projectile doesn't despawn when shooting next to wall at the same time.

**Comment by CJCrafter** [comment]
Kind of a "duh" moment, but why didn't we ever just make these armor stands markers?
